code/datums/status_effects/buffs.dm

```diff
@@ -188,6 +188,35 @@
 /datum/status_effect/blood_rush/on_remove()
 	REMOVE_TRAIT(owner, TRAIT_GOTTAGOFAST, VAMPIRE_TRAIT)
 
+/datum/status_effect/force_shield
+	id = "forceshield"
+	duration = 4 SECONDS
+	tick_interval = 0
+	var/mutable_appearance/shield
+
+/datum/status_effect/force_shield/on_apply()
+	. = ..()
+	if(!. || !ishuman(owner))
+		return
+	var/mutable_appearance/MA = mutable_appearance('icons/effects/effects.dmi', "shield-old", ABOVE_MOB_LAYER)
+	var/mob/living/carbon/human/H = owner
+	H.add_overlay(MA)
+	shield = MA
+	H.status_flags &= ~(CANSTUN | CANWEAKEN | CANPARALYSE | CANPUSH)
```

Comment:
the only solution I can think of without a major refactor, which I might end up doing, is cashing the users old status_flags, and then applying them on removal. this would still cause issues if they hit someone, then used hulk. if this was a trait rather than a bitflag, we could have it applied from multiple different sources and then remove them that way.

Reply:
made use of `stun_absorption`